src/polymer/declaration-property-handlers.ts

```diff
@@ -16,15 +16,48 @@ import * as estree from 'estree';
 
 import * as astValue from '../javascript/ast-value';
 import {JavaScriptDocument} from '../javascript/javascript-document';
-import {Severity} from '../warning/warning';
+import {ScannedBehaviorAssignment} from '../polymer/behavior';
+import {Severity, Warning} from '../warning/warning';
 
 import {analyzeProperties} from './analyze-properties';
 import {ScannedPolymerElement} from './polymer-element';
 
+export type BehaviorAssignmentOrWarning = {
+  kind: 'warning',
+  warning: Warning
+} | {kind: 'behaviorAssignment', assignment: ScannedBehaviorAssignment};
+
+export function getBehaviorAssignmentOrWarning(
```

Comment:
Those are good points! I wish TypeScript had the ability to describe the expected errors that may be thrown by a fn.